src/main/java/net/pms/newgui/components/CustomJSpinner.java

```diff
@@ -0,0 +1,315 @@
+				((SpinnerIntModel) model).getStepSize()
+			));
+		}
+
+		if (enterMoveFocus) {
+			if (this.getEditor() instanceof CustomJSpinner.IntegerEditor) {
+				((CustomJSpinner.IntegerEditor) this.getEditor()).getTextField().addKeyListener(new KeyListener() {
+					@Override
+					public void keyTyped(KeyEvent e) {}
+					@Override
+					public void keyReleased(KeyEvent e) {}
+					@Override
+					public void keyPressed(KeyEvent e) {
+						if (e.getKeyCode() == KeyEvent.VK_ENTER) {
+							FocusManager.getCurrentManager().focusNextComponent();
+						}
+					}
+				});
+			}
+		}
+	}
+
+	protected JComponent createEditor(SpinnerModel model) {
+		if (model instanceof SpinnerDateModel) {
+			return new DateEditor(this);
+		} else if (model instanceof SpinnerNumberModel) {
+			return new NumberEditor(this);
+		} else if (model instanceof SpinnerListModel) {
+			return new ListEditor(this);
+		} else if (model instanceof SpinnerIntModel) {
+			return new IntegerEditor(this);
+		} else {
+			return new DefaultEditor(this);
+		}
+	}
+
+	public JToolTip createToolTip() {
+	    JToolTip tip = new HyperLinkToolTip();
+	    tip.setComponent(this);
+	    return tip;
+	}
+
+	protected class MouseWheelRoll  implements MouseWheelListener {
+
+		private int minimum, maximum, stepSize;
+		private CustomJSpinner spinner;
+
+		private MouseWheelRoll(CustomJSpinner spinner, int minimum, int maximum, int stepSize) {
+			this.spinner = spinner;
+			this.minimum = minimum;
+			this.maximum = maximum;
+			this.stepSize = stepSize;
+		}
+
+		@Override
+		public void mouseWheelMoved(MouseWheelEvent e) {
+	        if (e.getScrollType() != MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+	            return;
+	        }
+
+	        int value = (int) spinner.getValue();
+	        value -= e.getWheelRotation() * stepSize;
+	        if (value == minimum + stepSize && minimum % stepSize != 0) {
+	        	value = ((minimum / stepSize) * stepSize) + stepSize;
+	        } else if (value == maximum - stepSize && maximum % stepSize != 0) {
+	        	value = (maximum / stepSize) * stepSize;
+	        }
+			value = Math.min(Math.max(value, minimum),maximum);
+	        spinner.setValue(value);
+		}
+	}
+
+    /**
+     * An editor for a <code>JSpinner</code> whose model is a
+     * <code>SpinnerIntModel</code>.  The value of the editor is
+     * displayed with a <code>JFormattedTextField</code> whose format
+     * is defined by a <code>NumberFormatter</code> instance whose
+     * <code>minimum</code> and <code>maximum</code> properties
+     * are mapped to the <code>SpinnerIntModel</code>.
+     */
+    public static class IntegerEditor extends DefaultEditor
+    {
+        // This is here until DecimalFormat gets a constructor that
+        // takes a Locale: 4923525
+        private static String getDefaultPattern(Locale locale) {
+            // Get the pattern for the default locale.
+            ResourceBundle rb = LocaleData.getNumberFormatData(locale);
```

Comment:
@skeptical Are you sure that that's the problem? I've been compiling both from within Eclipse and from maven command line, without problems. I know it's "bad practive" to use internal sun libs, and Eclipse actually creates a warning for it, but it's used several places in UMS (HTTPServer is one such class) already so I doubt that can be the reason. Since ```CustomJSpinner``` is just a subclassing of the swing component ```JSpinner```, I've used the same underlying libraries as they have (instead of "reinventing" the methods). There's a change in the underlying libs from Java 1.7 to Java 1.8, and you get that exact error if you try to compile the Java 6 or Java 7 (this PR) version with Java 8/JDK 1.8. #646 should compile fine under Java 8. 

Can you test if that is the problem? It's a bit strange though, as I thought Maven controlled the compiler version, but there's still much I don't understand about Maven.